### PR TITLE
Initialze RAPIDS Shuffle Manager at driver/executor startup

### PR DESCRIPTION
**Pull request by abellina**
Related to https://github.com/NVIDIA/spark-rapids/issues/3450. 

Given changes added to support Spark 3.2.0, a proxy system was put in place wrapping the RAPIDS Shuffle Manager. The proxy class has a `self` property that it uses to refer to the actual shuffle manager, the one that instantiates the transport. This all worked where the manager gets instantiated (for real) when the first shuffle is registered, or when `self` is used with early startup in the driver, but it was not convenient since it didn't print a message meant for a user to be able to turn off the shuffle, and also the executor didn't allocate memory until the first shuffle was registered, which it could have been doing all along.

This PR makes the initialization explicit, and hooks it in the driver/executor after we decide that the RAPIDS Shuffle Manager is configured.

### CODE REVIEW

**Comment by abellina** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Comment by tgravescs** [comment]
10:42:58  stderr: fatal: unable to access 'https://github.com/NVIDIA/spark-rapids.git/': The requested URL returned error: 429

**Comment by tgravescs** [comment]
build

**Review by jlowe — APPROVED** [review]
Can you elaborate a bit more in the PR description what this is doing?  Helps the software archeologists do a bit less digging later. 😄

**Comment by abellina** [comment]
> Can you elaborate a bit more in the PR description what this is doing? Helps the software archeologists do a bit less digging later.

Sorry, I thought (in my brain) I had written more, but I actually hadn't. Thanks for pointing it out.

**Review by gerashegalov — DISMISSED** [review]

**Comment by abellina** [comment]
There's a relevant test failure:
```
11:02:09  - Avoid transition to GPU for trivial projection after CPU SMJ *** FAILED ***
11:02:09    java.lang.IllegalStateException: Cannot initialize the RAPIDS Shuffle Manager
11:02:09    at org.apache.spark.sql.rapids.GpuShuffleEnv$.initShuffleManager(GpuShuffleEnv.scala:96)
11:02:09    at com.nvidia.spark.rapids.RapidsDriverPlugin.init(Plugin.scala:156)
11:02:09    at org.apache.spark.internal.plugin.DriverPluginContainer.$anonfun$driverPlugins$1(PluginContainer.scala:50)
11:02:09    at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:244)
11:02:09    at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
11:02:09    at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
11:02:09    at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
11:02:09    at scala.collection.TraversableLike.flatMap(TraversableLike.scala:244)
11:02:09    at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:241)
11:02:09    at scala.collection.AbstractTraversable.flatMap(Traversable.scala:108)
```

Looking into it.

**Comment by gerashegalov** [comment]
Try to avoid using GpuShuffleEnv, it's been quite difficult to work with in the prototype.

**Comment by abellina** [comment]
Yes @gerashegalov, that's right. The lazy vals in `GpuShuffleEnv` are the root of this, where test X configures the shuffle manager, but test Y doesn't want that enabled.

Actually this is a latent bug... I added an exception here, which is exposing the issue. Looking into a fix.

**Comment by abellina** [comment]
@gerashegalov I restored a step that was removed with the shim rework for 3.2, where `shutdown` in `GpuShuffleEnv` would null out `env`. Please take another look when you get a chance.

**Comment by abellina** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuShuffleEnv.scala [hunk 1]
```diff
@@ -84,6 +88,19 @@ object GpuShuffleEnv extends Logging {
     SparkEnv.get.blockManager.externalShuffleServiceEnabled
   }
 
+  //
+  // The actual instantiation of the RAPIDS Shuffle Manager is lazy, and
+  // this forces the initialization when we know we are ready in the driver and executor.
+  //
+  def initShuffleManager(): Unit = {
+    SparkEnv.get.shuffleManager match {
+      case visibleMgr: VisibleShuffleManager=>
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuShuffleEnv.scala:97]:
```suggestion
      case visibleMgr: VisibleShuffleManager =>
```

**Review by jlowe — APPROVED** [review]
Small whitespace nit but otherwise lgtm.

**Comment by abellina** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**